pkg/controllers/common/ingress_config.go

```diff
@@ -28,3 +28,36 @@ func GetIngressConfig(ingressLister configv1lister.IngressLister, conditionPrefi
 	}
 	return ingress, nil
 }
+
+// getComponentRouteSpec searches the entries of the ingress.spec.componentRoutes array for a componentRoute with a matching namespace and name.
+// If a componentRoute is found it is returned along with a true boolean
+// If a componentRoute is not found an empty component route is return along with a false.
+func getComponentRouteSpec(ingress configv1.Ingress, namespace string, name string) (configv1.ComponentRouteSpec, bool) {
+	for _, componentRoute := range ingress.DeepCopy().Spec.ComponentRoutes {
```

Comment:
Looks good Alex.
I always go with the short form on parameters ex: `(ingress configv1.Ingress, namespace, name string)` but this is mostly a style preference